api/v3/Domain.php

```diff
@@ -129,8 +133,19 @@ function _civicrm_api3_domain_get_spec(&$params) {
  * @return array
  */
 function civicrm_api3_domain_create($params) {
-  $params['version'] = $params['domain_version'];
-  return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (isset($params['domain_version'])) {
+    $params['version'] = $params['domain_version'];
+  }
+  else {
+    unset($params['version']);
+  }
+  $result = _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+
+  // Rename version to domain_version, see CRM-17430.
+  $result_value = CRM_Utils_Array::first($result['values']);
+  $result_value['domain_version'] = $result_value['version'];
+  unset($result_value['version']);
+  return $result;
```

Comment:
I don't see how the above four lines of code could possibly work. CRM_Utils_Array::first does not return a reference.